shell/renderer/api/electron_api_context_bridge.cc

```diff
@@ -408,29 +408,39 @@ void ProxyFunctionWrapper(const v8::FunctionCallbackInfo<v8::Value>& info) {
 
     v8::MaybeLocal<v8::Value> maybe_return_value;
     bool did_error = false;
-    std::string error_message;
+    v8::Local<v8::Value> error_message;
     {
       v8::TryCatch try_catch(args.isolate());
       maybe_return_value = func->Call(func_owning_context, func,
                                       proxied_args.size(), proxied_args.data());
       if (try_catch.HasCaught()) {
         did_error = true;
-        auto message = try_catch.Message();
-
-        if (message.IsEmpty() ||
-            !gin::ConvertFromV8(args.isolate(), message->Get(),
-                                &error_message)) {
-          error_message =
-              "An unknown exception occurred in the isolated context, an error "
-              "occurred but a valid exception was not thrown.";
+        v8::Local<v8::Value> exception = try_catch.Exception();
+
+        const char* err_msg =
+            "An unknown exception occurred in the isolated context, an error "
+            "occurred but a valid exception was not thrown.";
+
+        if (!exception->IsNull() && exception->IsObject()) {
+          v8::MaybeLocal<v8::Value> maybe_message =
+              exception.As<v8::Object>()->Get(
+                  func_owning_context,
+                  gin::ConvertToV8(args.isolate(), "message"));
```

Comment:
ehhhh, I do not love this, as `Object::Get` can call arbitrary JS and end up throwing again. Can we find a different way to trim the message?

Reply:
Yeah... I'm also confused by
> errors aren't serializable as-is

Error is definitely serializable 🤔

Reply:
@MarshallOfSound do you know why it was added that way initially?

Reply:
@nornagon this looks pretty similar to what V8 themselves do: https://source.chromium.org/chromium/chromium/src/+/master:v8/src/inspector/value-mirror.cc;l=245-253;drc=effdbe12c7d315055a0a4ddfea3770f2dc2cf84d?q=%22value-%3EIsSymbol()%22&ss=chromium

Reply:
huh, TIL. That's in inspector code, so not really prod stuff, but it's good enough for me.